webnode/src/components/storage/index.js

```diff
@@ -73,7 +74,29 @@ class Storage extends Component {
       }))
       .map(obj => findTreasure(obj));
 
-    Promise.all(transformedMap); //TODO better implementation
+    chilloutForEach([1], () => {
+      Promise.all(transformedMap); //TODO better implementation
+    }).then(() => {});
+  }
+
+  async onClickWorker() {
+    const { workerFindTreasure } = this.props;
+
+    const generatedMap = datamap.generate(
+      this.state.genesisHash,
+      this.state.numberOfChunks
+    );
+
+    const transformedMap = _.toArray(generatedMap)
+      .map((value, index) => ({
+        dataMapHash: value,
+        chunkIdx: index
+      }))
+      .map(obj => workerFindTreasure(obj));
+
+    chilloutForEach([1], () => {
+      Promise.all(transformedMap); //TODO better implementation
+    }).then(() => {});
   }
```

Comment:
If I'm understanding this PR correctly, basically what you're doing is duplicating a lot of the logic that we already do, except moving it into Chillout?  And once we verify it works, we'd remove the old logic so it's only the stuff that's inside Chillout?

Reply:
chiilout is basicly cycling code and inside it is managing performance